src/index.ts

```diff
@@ -113,6 +117,49 @@ async function serverChannel(
     }
   });
 
+  let started = false;
+  localBuildProgress.on("change", (progress) => {
+    if (progress?.currentStep === "upload" && progress.storybookBuildDir && !started) {
+      console.log("Build step completed, starting local extract");
+      const app = express();
+      app.get("/extract.html", (request, response) => {
+        response.writeHead(200, { contentType: "text/html" });
+        const content = dedent`
+          <iframe src="/iframe.html"></iframe>
+          <script>
+            const iframe = window.frames[0];
+
+            iframe.onload = async () => {
+              const preview = iframe.__STORYBOOK_PREVIEW__;
+              await preview.storyStore.initializationPromise;
+              const extracted = await preview.extract();
+
+              const chromaticParameters = Object.fromEntries(
+                Object.entries(extracted).map(([storyId, { parameters: { chromatic }}]) => 
+                  [storyId, { parameters: { chromatic } }]
+                )
+              );
```

Comment:
This a very rough approximation of what our extract script does right now, but it's sufficient to allow the data to be passed over the post message channel.